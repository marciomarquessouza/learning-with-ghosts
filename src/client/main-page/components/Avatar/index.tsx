import React from 'react'

import { getInitials } from '../../utils'

interface AvatarProps {
    name: string
    photoUrl?: string | null
}

function Avatar({ name, photoUrl }: AvatarProps) {
    return (
        <div>
            {photoUrl ? (
                <img
                    className="inline-block h-12 w-12 rounded-full"
                    referrerPolicy={'no-referrer'}
                    src={photoUrl}
                    alt=""
                />
            ) : (
                <span className="inline-flex h-12 w-12 items-center justify-center rounded-full bg-gray-500">
                    <span className="text-lg font-medium leading-none text-white">
                        {getInitials(name)}
                    </span>
                </span>
            )}
        </div>
    )
}

export default Avatar
