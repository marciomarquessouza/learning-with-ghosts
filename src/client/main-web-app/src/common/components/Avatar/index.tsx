import React from 'react'
import Image from 'next/image'

import { getInitials } from '../../../utils'

interface AvatarProps {
    name: string
    photoUrl?: string | null
}

function Avatar({ name, photoUrl }: AvatarProps) {
    return (
        <div>
            {photoUrl ? (
                <Image
                    className="inline-block h-12 w-12 rounded-full"
                    referrerPolicy={'no-referrer'}
                    src={photoUrl}
                    alt="User Avatar"
                    width={200}
                    height={200}
                />
            ) : (
                <span className="inline-flex h-12 w-12 items-center justify-center rounded-full bg-gray-500">
                    <span className="text-lg font-medium leading-none text-white">
                        {getInitials(name)}
                    </span>
                </span>
            )}
        </div>
    )
}

export default Avatar
