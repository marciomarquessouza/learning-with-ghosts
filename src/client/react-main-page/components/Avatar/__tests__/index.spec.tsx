import React from 'react'
import { render } from '@testing-library/react'

import Avatar from '..'

describe('components >> Avatar', () => {
    it('should render the Avatar component properly', () => {
        const mockName = '_MOCK_NAME_'
        const mockUrl = '_MOCK_URL_'
        const { container } = render(<Avatar name={mockName} photoUrl={mockUrl} />)

        expect(container).toMatchSnapshot()
    })
})
