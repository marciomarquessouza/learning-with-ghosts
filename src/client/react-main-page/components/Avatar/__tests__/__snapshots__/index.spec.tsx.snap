// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> Avatar should render the Avatar component properly 1`] = `
<div>
  <div>
    <img
      alt=""
      class="inline-block h-12 w-12 rounded-full"
      referrerpolicy="no-referrer"
      src="_MOCK_URL_"
    />
  </div>
</div>
`;
